src/rpc/net.cpp

```diff
@@ -509,6 +509,64 @@ static RPCHelpMan getaddednodeinfo()
     };
 }
 
+UniValue AggregateNetMsgStats(const std::vector<StatsFilter>& filters, const CConnman::Stats& raw_stats)
+{
+    const size_t num_filters = filters.size();
+    size_t map_depth = 1;
+
+    if (num_filters > 0) {
+        map_depth = num_filters;
+    }
+
+    MultiLevelStatsMap result{map_depth};
+
+    // Iterate over the multi dimensional array that holds the raw stats
+    for (int network_index = 0; network_index < NET_MAX; network_index++) {
+        for (std::size_t connection_index = 0; connection_index < static_cast<std::size_t>(ConnectionType::NUM_CONN_TYPES); connection_index++) {
+            for (std::size_t message_index = 0; message_index < (NUM_NET_MESSAGE_TYPES + 1); message_index++) {
+                const CConnman::MsgStatsValue& raw_stats_value = raw_stats[network_index][connection_index][message_index];
+                std::vector<std::string> path{};
+                // "result" is our multi-level map, we use path to keep track of where we are as we move down it
+
+                std::string filter_name = "";
+
+                for (unsigned int i = 0; i < num_filters; i++) {
+                    StatsFilter filter = filters[i];
+
+                    // get the filter_name, which tells us which subtree to move into
+                    if (filter == StatsFilter::MESSAGE_TYPE) {
+                        if (message_index == NUM_NET_MESSAGE_TYPES) {
+                            filter_name = "other"; // instead of *other*
```

Comment:
Would be easier if an `other` message type was added to the `allNetMessageTypes` array in `protocol.cpp`?